### DISCUSSION TITLE
certificate creation error

### DISCUSSION THREAD

**Discussion by ifemi** [cloud-ark/kubeplus#954] [Q&A]
Hello,

I get the following error while trying to deploy on Minikube on WSL with docker desktop:

unable to recognize "STDIN": no matches for kind "CertificateSigningRequest" in version "certificates.k8s.io/v1beta1"

kindly help

**Comment by devdattakulkarni** [comment]
@ifemi Can you share following information:
```
kubectl explain CertificateSigningRequest
```
Also,
```
kubectl version
```
And the KubePlus installation command.

**Reply by ifemi** [reply]
kubectl explain CertificateSigningRequest
KIND:     CertificateSigningRequest
VERSION:  certificates.k8s.io/v1

DESCRIPTION:
     CertificateSigningRequest objects provide a mechanism to obtain x509
     certificates by submitting a certificate signing request, and having it
     asynchronously approved and issued.

     Kubelets use this API to obtain:
     1. client certificates to authenticate to kube-apiserver (with the
     "kubernetes.io/kube-apiserver-client-kubelet" signerName).
     2. serving certificates for TLS endpoints kube-apiserver can connect to
     securely (with the "kubernetes.io/kubelet-serving" signerName).

     This API can be used to request client certificates to authenticate to
     kube-apiserver (with the "kubernetes.io/kube-apiserver-client" signerName),
     or to obtain certificates from custom non-Kubernetes signers.

FIELDS:
   apiVersion   <string>
     APIVersion defines the versioned schema of this representation of an
     object. Servers should convert recognized schemas to the latest internal
     value, and may reject unrecognized values. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources

   kind <string>
     Kind is a string value representing the REST resource this object
     represents. Servers may infer this from the endpoint the client submits
     requests to. Cannot be updated. In CamelCase. More info:
     https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds

   metadata     <Object>

   spec <Object> -required-
     spec contains the certificate request, and is immutable after creation.
     Only the request, signerName, expirationSeconds, and usages fields can be
     set on creation. Other fields are derived by Kubernetes and cannot be
     modified by users.

   status       <Object>
     status contains information about whether the request is approved or
     denied, and the certificate issued by the signer, or the failure condition
     indicating signer failure.


kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5", GitCommit:"aea7bbadd2fc0cd689de94a54e5b7b758869d691", GitTreeState:"clean", BuildDate:"2021-09-15T21:10:45Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:35:25Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}

**Reply by ifemi** [reply]
@devdattakulkarni , below is the kubeplus installation command used:

KUBEPLUS_NS=default
./deploy-kubeplus.sh $KUBEPLUS_NS

Many thanks

**Reply by devdattakulkarni** [reply]
@ifemi: Thanks!

I would suggest using Kubernetes version (server and client) 1.19. In Kubernetes version 1.19+ api groups for many of the Kubernetes resources have changed. We have not yet migrated KubePlus to accommodate this change. So till we do that use Kubernetes version 1.19.

Also, please use Helm command from the main README to install KubePlus instead of the deploy-kubeplus script.

```
$ KUBEPLUS_NS=default
$ helm install kubeplus "https://github.com/cloud-ark/operatorcharts/blob/master/kubeplus-chart-2.0.6.tgz?raw=true" -n $KUBEPLUS_NS
```

**Reply by ifemi** [reply]
Ok. I will give it a shot and revert @devdattakulkarni. Many thanks

**Reply by devdattakulkarni** [reply]
@ifemi 👍

**Reply by ifemi** [reply]
Hello @devdattakulkarni  switching to kubernetes v1.19 and using the helm install worked.

Great job. Many thanks

I ran into another issue but I will ask that as a different question

**Reply by devdattakulkarni** [reply]
@ifemi Glad to know that switching to kubernetes v1.19 worked.
Sure, let us know what issue that you are running into. If it is a error, I would just suggest creating Github issue instead of a question.

**Reply by ifemi** [reply]
> @ifemi Glad to know that switching to kubernetes v1.19 worked. Sure, let us know what issue that you are running into. If it is a error, I would just suggest creating Github issue instead of a question.

Ok. I will do that.
Cheers
